packages/file_selector/file_selector_windows/lib/src/dart_file_selector_api.dart

```diff
@@ -109,6 +110,34 @@ class DartFileSelectorAPI extends FileDialog {
     return hResult;
   }
 
+  /// Sets the initial directory to open the dialog
+  @visibleForTesting
+  int setInitialDirectory(
+      int hResult, String? initialDirectory, IFileOpenDialog dialog) {
+    if (initialDirectory == null || initialDirectory.isEmpty) {
+      return hResult;
+    }
+
+    using((Arena arena) {
+      final Pointer<GUID> guid = GUIDFromString(IID_IShellItem);
+      final Pointer<Pointer<COMObject>> dirPath = arena<Pointer<COMObject>>();
+      hResult = SHCreateItemFromParsingName(
+          TEXT(initialDirectory), nullptr, guid, dirPath);
+
+      if (FAILED(hResult)) {
+        throw WindowsException(hResult);
+      }
+
+      hResult = dialog.setFolder(dirPath.value);
```

Comment:
Have you seen the abstraction we are using to interact with Dialogs?

Reply:
Oh I've just seen that way. Do you mean to move the setFolder to fileselectorapi.setInitialDirectory ?